src/kubernetes.cc

```diff
@@ -843,6 +903,41 @@ void KubernetesReader::WatchNodes(
   LOG(INFO) << "Watch thread (node) exiting";
 }
 
+void KubernetesReader::ResourceCallback(
+    MetadataUpdater::UpdateCallback callback,
+    const json::Object* resource, Timestamp collected_at, bool is_deleted) const
+    throw(json::Exception) {
+  std::vector<MetadataUpdater::ResourceMetadata> result_vector;
+  result_vector.emplace_back(
+      GetResourceMetadata(resource, collected_at, is_deleted));
+  callback(std::move(result_vector));
+}
+
+void KubernetesReader::WatchResources(
```

Comment:
Seems like this is now very similar to `WatchPods` and `WatchNodes`. Can we refactor the code to reuse this? E.g., all of the API path construction could be pulled out into the watch thread lambda.

Reply:
sure. working on this.